hrjobcontract/tests/phpunit/CRM/Hrjobcontract/RevisionEffectiveDateTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+require_once 'CiviTest/CiviUnitTestCase.php';
+
+/**
+ * Class CRM_Hrjobcontract_RevisionEffectiveDateTest
+ */
+class CRM_Hrjobcontract_RevisionEffectiveDateTest extends CiviUnitTestCase {
+  private $_time = null;
+  private $_testContactId = null;
+  private $_testJobContractId = null;
+
+  function setUp() {
+    parent::setUp();
+    $this->quickCleanup(array(
+      'civicrm_hrjobcontract_details',
+      'civicrm_hrjobcontract_health',
+      'civicrm_hrjobcontract_role',
+      'civicrm_hrjobcontract_hour',
+      'civicrm_hrjobcontract_pay',
+      'civicrm_hrjobcontract_leave',
+      'civicrm_hrjobcontract_pension',
+      'civicrm_hrjobcontract_revision',
+      'civicrm_hrjobcontract',
+    ));
+    $upgrader = CRM_Hrjobcontract_Upgrader::instance();
+    $upgrader->install();
+    $this->_time = time();
+    $this->_createTestContact();
+    $this->_createTestJobContract($this->_testContactId);
+  }
+
+  function tearDown() {
+    parent::tearDown();
+  }
+
+  function testRevisionEffectiveDates() {
```

Comment:
What about splitting this test method into smaller ones with more descriptive names? The idea is that each test method would test only a single scenario. Just one example:
```php
public function testIfNewEffectiveDateIsEqualsToPreviousEffectiveDateThenItShouldBeOverridden() {
...
}
```